compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/RegExpLexer.java

```diff
@@ -56,43 +55,11 @@ public STToken nextToken() {
             case RE_UNICODE_PROP_ESCAPE:
                 token = readTokenInReUnicodePropertyEscape();
                 break;
-            case RE_UNICODE_PROP_START:
-                token = readTokenInReUnicodePropertyStart();
-                break;
             case RE_UNICODE_PROPERTY_VALUE:
                 token = readTokenInReUnicodePropertyValue();
                 break;
-            case RE_UNICODE_GENERAL_CATEGORY_NAME:
-                token = readTokenInReUnicodeGeneralCategoryName();
-                break;
-            case RE_CHAR_SET:
-            case RE_NEGATED_CHAR_CLASS_START:
-                token = readTokenInCharacterClass();
-                break;
-            case RE_CHAR_SET_RANGE_RHS:
-            case RE_CHAR_SET_RANGE_RHS_START:
-                token = readTokenInReCharSetRangeRhs();
-                break;
-            case RE_CHAR_SET_NO_DASH:
-                token = readTokenInReCharSetNoDash();
-                break;
-            case RE_CHAR_SET_RANGE_NO_DASH_RHS_START:
-            case RE_CHAR_SET_RANGE_NO_DASH_RHS:
-                token = readTokenInReCharSetRangeNoDashRhs();
-                break;
-            case RE_FLAGS_START:
-            case RE_FLAGS:
-                token = readTokenInFlagsExpression();
-                break;
-            case RE_QUANTIFIER:
-            case RE_NON_GREEDY_QUANTIFIER:
-                token = readTokenInQuantifier();
-                break;
-            case RE_BRACED_QUANTIFIER:
-            case RE_BRACED_QUANTIFIER_LEAST_DIGITS:
-            case RE_BRACED_QUANTIFIER_MOST_DIGITS_START:
-            case RE_BRACED_QUANTIFIER_MOST_DIGITS:
-                token = readTokenInBracedQuantifier();
+            case RE_FLAG_EXPRESSION:
+                token = readTokenInFlagExpression();
```

Comment:
Similarly, we can remove the `RE_FLAG_EXPRESSION` mode and let it be tokenized in the `RE_DISJUNCTION` mode and ,then in the parser, tokens: `-`, `:`, `m`,`s`, `i`, `x` captured as `RE_LITERAL_CHAR` are converted to the required one. The idea is to avoid having parser mode as much as possible to improve readability and bring the contextual differences in the token kinds to the parser level.